c10/core/impl/CUDATraceTLS.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <c10/core/impl/PyInterpreter.h>
+
+namespace c10 {
+namespace impl {
+
+static constexpr char trace_cuda_event_creation_fn_name[] =
+  "CUDAEventCreationCallbacks";
+static constexpr char trace_cuda_event_deletion_fn_name[] =
+  "CUDAEventDeletionCallbacks";
+static constexpr char trace_cuda_event_record_fn_name[] =
+  "CUDAEventRecordCallbacks";
+static constexpr char trace_cuda_event_wait_fn_name[] =
+  "CUDAEventWaitCallbacks";
+static constexpr char trace_cuda_memory_allocation_fn_name[] =
+  "CUDAMemoryAllocationCallbacks";
+static constexpr char trace_cuda_memory_deallocation_fn_name[] =
+  "CUDAMemoryDeallocationCallbacks";
+static constexpr char trace_cuda_stream_allocation_fn_name[] =
+  "CUDAStreamAllocationCallbacks";
```

Comment:
Do these constants have to live here? They seem to be only used in python_variable.cpp, so maybe it is better to move them closer to where they are used? That would also allow them to be only included in the .cpp file, instead of in a header file, which means less complexity (and quicker parsing!) whenever someone is including this header (directly or recursively). In general one should be mindful to put as little as possible in headers.

Reply:
I've moved them to `python_variable.cpp` as you said. I guess this file is also better because it is the one directly hooking into Python, so it's "closest" to the place where the CallbackRegistry objects are defined.